common.py

```diff
@@ -733,12 +765,18 @@ def get_fisher_bao(self) :
                     dda_nodes[i,:]=(-3*da_fid_nodes+4*dam-dap)/(2*sig*self.params_fshr[i].dval)
                     dhh_nodes[i,:]=(-3*hh_fid_nodes+4*hhm-hhp)/(2*sig*self.params_fshr[i].dval)
                     ddv_nodes[i,:]=(-3*dv_fid_nodes+4*dvm-dvp)/(2*sig*self.params_fshr[i].dval)
-                dda_nodes[i,:]/=da_fid_nodes
-                dhh_nodes[i,:]/=hh_fid_nodes
-                ddv_nodes[i,:]/=dv_fid_nodes
+
+                if self.relative_bao_errors:
+
+                    dda_nodes[i,:]/=da_fid_nodes
+                    dhh_nodes[i,:]/=hh_fid_nodes
+                    ddv_nodes[i,:]/=dv_fid_nodes
+
             self.fshr_bao+=np.sum(dda_nodes[:,None,:]*dda_nodes[None,:,:]/self.e_nodes_DA**2,axis=2)
             self.fshr_bao+=np.sum(dhh_nodes[:,None,:]*dhh_nodes[None,:,:]/self.e_nodes_HH**2,axis=2)
             self.fshr_bao+=np.sum(ddv_nodes[:,None,:]*ddv_nodes[None,:,:]/self.e_nodes_DV**2,axis=2)
+            np.savetxt(fname_save,self.fshr_bao)
```

Comment:
Note that in this new version all the data (total fisher, fisher for power spectra, fisher for BAO etc.) are saved in the same file, so one shouldn't just save the BAO fisher in this file. This is done above in "join_fishers", which gets called at the end of the run